src/components/NetworkBN/component.js

```diff
@@ -0,0 +1,192 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { nodePropTypes, networkPropTypes, inferenceResultsPropTypes } from 'models';
+import { getArrowsPositions } from 'utils/arrows-positions';
+import Network from '../Network';
+
+import AddNodeModal from '../AddNodeModal';
+import EditCptModal from '../EditCptModal';
+import EditStatesModal from '../EditStatesModal';
+
+class NetworkBN extends Component {
+  state = {
+    editingNodeStates: null,
+    editingNodeCpt: null,
+  };
+
+  componentDidMount() {
+    window.addEventListener('keyup', this.handleKeyup);
+  }
+
+  componentWillUnmount() {
+    window.removeEventListener('keyup', this.handleKeyup);
+  }
+
+  get nodes() {
+    const { inferenceResults, network, nodes } = this.props;
+
+    return nodes.map(node => ({
+      ...node,
+      name: node.id,
+      results: inferenceResults[node.id],
+      belief: network.beliefs[node.id],
+    }));
+  }
+
+  get arrowContextItems() {
+    const { onRemoveNodeConnection } = this.props;
+
+    return [
+      {
+        key: 'remove-link',
+        text: 'Remover ligação',
+        style: { color: '#C62828' },
+        onClick: (_, contextMenuArrow) => onRemoveNodeConnection(contextMenuArrow),
+      },
+    ];
+  }
+
+  get networkContextItems() {
+    return [
+      {
+        key: 'add-node',
+        text: 'Adicionar variável',
+        onClick: ({ mousePosition }) => {
+          this.net.createNode(mousePosition);
+        },
+      },
+    ];
+  }
+
+  get nodeContextItems() {
+    const { onRemoveNode } = this.props;
+
+    return [
+      {
+        key: 'add-child',
+        text: 'Adicionar ligação',
+        onClick: (_, contextMenuNode) => {
+          this.net.startConnection(contextMenuNode);
+        },
+      },
+      {
+        key: 'edit-states',
+        text: 'Editar estados',
+        onClick: (_, contextMenuNode) => {
+          this.onEditNodeStates(contextMenuNode);
+        },
+      },
+      {
+        key: 'edit-cpt',
+        text: 'Editar probabilidades',
+        onClick: (_, contextMenuNode) => {
+          this.onEditNodeCpt(contextMenuNode);
+        },
+      },
+      {
+        key: 'remove-node',
+        text: 'Remover variável',
+        style: { color: '#C62828' },
+        onClick: (_, contextMenuNode) => onRemoveNode(contextMenuNode),
+      },
+    ];
+  }
+
+  onEditNodeStates = (editingNodeStates) => {
+    this.setState({ editingNodeStates });
+  };
+
+  onEditNodeCpt = (editingNodeCpt) => {
+    this.setState({ editingNodeCpt });
+  };
+
+  handleKeyup = (e) => {
+    const key = e.keyCode || e.which;
+    const { network, onRemoveNode } = this.props;
+
+    if ([8, 46].indexOf(key) !== -1 && network.selectedNodes.length > 0 && document.activeElement.tagName === 'BODY') {
```

Comment:
You could use includes here:

```suggestion
    if ([8, 46].includes(key) && network.selectedNodes.length > 0 && document.activeElement.tagName === 'BODY') {
```
Also, I would assign these numbers to constants to be clearer about them, like:
```
const keyUp = 8;
```